File: Bachelor/Semester4/Systems_for_Design_and_Implementation/C#/ProiectCShMPP/Networking/Utils/AbstractConcurrentServer.cs
using System.Net.Sockets;
using System.Threading;

namespace Networking.Utils
{
    public abstract class AbstractConcurrentServer : AbstractServer
    {
        protected AbstractConcurrentServer(string ip, int port) : base(ip, port)
        {
        }

        protected override void ProcessRequest(TcpClient client)
        {
            CreateWorker(client).Start();
        }

        protected abstract Thread CreateWorker(TcpClient client);
    }
}

File: Bachelor/Semester5/Parallel_and_Distributed_Programming/P1/Networking/src/main/java/utils/AbstractConcurrentServer.java
package utils;

import java.net.Socket;

public abstract class AbstractConcurrentServer extends AbstractServer {
    protected AbstractConcurrentServer(int port) {
        super(port);
    }

    @Override
    protected void processRequest(Socket client) {
        createWorker(client);
    }

    protected abstract void createWorker(Socket client);
}
